Deferred/FutureCollections.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FutureCollections.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 8/29/15.
+//  Copyright © 2014-2015 Big Nerd Ranch. Licensed under MIT.
+//
+
+import Dispatch
+
+public extension SequenceType where Generator.Element: FutureType {
+    /// Choose the future that is determined first from a collection of futures.
+    ///
+    /// - returns: A deferred value that is determined with the first of the
+    ///   given futures to be determined.
+    var firstFuture: Deferred<Generator.Element.Value> {
```

Comment:
#20 doesn't include any discussion of verbosity problems. I think you have a mental draft that needs posting. ;)